src/app/games/pylos/PylosCoord.ts

```diff
@@ -1,43 +1,23 @@
 import { MGPOptional } from 'src/app/utils/MGPOptional';
 import { Coord } from 'src/app/jscaip/Coord';
 import { Orthogonal } from 'src/app/jscaip/Direction';
+import { Coord3D } from 'src/app/jscaip/Coord3D';
+import { Encoder } from 'src/app/utils/Encoder';
+import { getOptionalEncoder } from 'src/app/utils/MGPOptionalEncoder';
 
-export class PylosCoord extends Coord {
+export class PylosCoord extends Coord3D {
 
-    public static encodeOptional(optionalCoord: MGPOptional<PylosCoord>): number {
-        let result: number;
-        if (optionalCoord.isPresent()) {
-            result = PylosCoord.encode(optionalCoord.get()) + 1; // From 1 to 64
-        } else {
-            result = 0;
-        } // result from 0 to 64
-        return result;
-    }
-    public static encode(coord: PylosCoord): number {
-        const z: number = coord.z;
-        const y: number = coord.y * 4;
-        const x: number = coord.x * 16;
-        return x + y + z; // from 0 to 63
-    }
-    public static decodeToOptional(encodedOptional: number): MGPOptional<PylosCoord> {
-        if (encodedOptional === 0) {
-            return MGPOptional.empty();
-        }
-        return MGPOptional.of(PylosCoord.decode(encodedOptional - 1));
-    }
-    public static decode(coord: number): PylosCoord {
-        const z: number = coord % 4;
-        coord -= z; coord /= 4;
+    public static encoder: Encoder<PylosCoord> = Coord3D.getEncoder(PylosCoord.from);
 
-        const y: number = coord % 4;
-        coord -= y; coord /= 4;
+    public static optionalEncoder: Encoder<MGPOptional<PylosCoord>> = getOptionalEncoder(PylosCoord.encoder);
 
-        const x: number = coord;
+    public static MAX_COORD: Coord3D = new Coord3D(3, 3, 3); // Coord3D because (3, 3, 3) is not be legal for Pylos
```

Comment:
ce commentaire n'a aucun sens ? :D

Reply:
// Cannot be typed as PylosCoord caue (3, 3, 3) is not a valid PylosCoord
AUrais été mieux, ceci dit c'est surtout une variable inutilisée !!!